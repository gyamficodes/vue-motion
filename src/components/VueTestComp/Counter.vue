  

<template>
    <div>
        <h1 class=" text-[40px] text-red-400">{{ counter }}</h1>
        <div class=" flex items-center gap-4">
          <button @click="handdleCounterDecreament">Decrease</button>
          <button @click="handdleCounterReset">Reset</button>
          <button @click="handdleCounterIncreament">Increase</button>
        </div>
    </div>
</template>

<script setup lang="ts">
import { ref } from 'vue';


const counter = ref<number>(0)

const  handdleCounterIncreament = () => counter.value++;
const  handdleCounterDecreament = () => counter.value--;
const  handdleCounterReset = () => counter.value = 0;


</script>

<style scoped>

</style>

// "type": "module",