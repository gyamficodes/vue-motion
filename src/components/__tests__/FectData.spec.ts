import { mount } from "@vue/test-utils";
import { describe, expect, test } from "vitest";
import Counter from "../VueTestComp/Counter.vue";

describe("Counter Component", () => {
  test("matches snapshot", () => {
    const wrapper = mount(Counter);
    expect(wrapper.html()).toMatchSnapshot();
  });
});