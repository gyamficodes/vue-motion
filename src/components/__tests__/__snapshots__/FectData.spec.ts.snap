// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter Component > matches snapshot 1`] = `
"<div>
  <h1 class="text-[40px] text-red-400">0</h1>
  <div class="flex items-center gap-4"><button>Decrease</button><button>Reset</button><button>Increase</button></div>
</div>"
`;



